### DISCUSSION TITLE
Arbitrary action input

### DISCUSSION THREAD

**Discussion by Xophmeister** [ghostty-org/ghostty#4195] [Feature Requests, Ideas] [answered]
In tmux, `Prefix+:` will open a command line input to allow you to enter arbitrary tmux commands. This is similar to Vim, where `:` enters command line mode.

It would be useful to have such a feature in Ghostty to try out actions, without having to define a keybinding for them.

**Comment by pluiedev** [comment]
This idea could work, but I ponder about the UI/UX of this. IMO it would be much better if we have an IntelliJ- or Blender-esque global menu where you can select any action to perform, or type in text that help you narrow down to the specific action you want. Using a text interface feels a bit archaic.

**Reply by Xophmeister** [reply]
> Using a text interface feels a bit archaic.

For a terminal emulator? Having alternative methods is fine, but as a heavy terminal user, it stands to reason that everything is doable from the keyboard.

**Reply by pluiedev** [reply]
Actually, a global menu popup like what I described is usually triggered by a keybind (e.g. ctrl+shift+a) too. I'm just not a big fan of actually embedding a text interface inside the existing terminal.

**Reply by zajca** [reply]
Global menu would be great, I'm lazy to remember what commands exists and shortcuts. Ctrl+shift+a and shift+shift for search everywhere are great thing in IntelliJ, or ctrl+shift+p in vscode.
https://code.visualstudio.com/assets/docs/getstarted/tips-and-tricks/OpenCommandPalette.gif

It would be weird to control UI of Ghostty from inside the actual command line, that would be similar to control vim behavior from edit mode.
You have also GUI tab overview, this is similar functionality.

**Reply by pluiedev** [reply, marked as answer]
Making this an actual issue as people (including maintainers) evidently are pretty supportive of the idea.
